contracts/SDCollateral.sol

```diff
@@ -51,21 +54,67 @@ contract SDCollateral is ISDCollateral, AccessControlUpgradeable, ReentrancyGuar
         emit SDDeposited(operator, _sdAmount);
     }
 
+    function depositUtilizedSD(address _operator, uint256 _sdAmount) external override {
+        UtilLib.onlyStaderContract(msg.sender, staderConfig, staderConfig.SD_UTILITY_POOL());
+
+        operatorUtilizedSDBalance[_operator] += _sdAmount;
+        if (!IERC20(staderConfig.getStaderToken()).transferFrom(msg.sender, address(this), _sdAmount)) {
+            revert SDTransferFailed();
+        }
+
+        emit UtilizedSDDeposited(_operator, _sdAmount);
+    }
+
+    function reduceUtilizedSDPosition(address operator, uint256 sdAmount) external override {
+        UtilLib.onlyStaderContract(msg.sender, staderConfig, staderConfig.SD_UTILITY_POOL());
+
+        operatorUtilizedSDBalance[operator] -= sdAmount;
+        operatorSDBalance[operator] += sdAmount;
+
+        emit ReducedUtilizedPosition(operator, sdAmount);
+    }
+
     /// @notice for operator to withdraw their sd collateral, which is over and above withdraw threshold
     function withdraw(uint256 _requestedSD) external override {
         address operator = msg.sender;
-        uint256 opSDBalance = operatorSDBalance[operator];
+        uint256 operatorUtilizedSD = operatorUtilizedSDBalance[operator];
+        uint256 opSDBalance = operatorSDBalance[operator] + operatorUtilizedSD;
 
         if (opSDBalance < getOperatorWithdrawThreshold(operator) + _requestedSD) {
             revert InsufficientSDToWithdraw(opSDBalance);
         }
-        operatorSDBalance[operator] -= _requestedSD;
+        uint256 operatorCurrentUtilizeSD = ISDUtilityPool(staderConfig.getSDUtilityPool()).utilizerBalanceCurrent(
+            operator
+        );
 
-        // cannot use safeERC20 as this contract is an upgradeable contract, and using safeERC20 is not upgrade-safe
-        if (!IERC20(staderConfig.getStaderToken()).transfer(payable(operator), _requestedSD)) {
-            revert SDTransferFailed();
+        uint256 utilizePositionChange = operatorUtilizedSD >= _requestedSD ? _requestedSD : operatorUtilizedSD;
+        operatorUtilizedSDBalance[operator] -= utilizePositionChange;
```

Comment:
```
SDRequestedLeft = SDRequested;
interestAccrued = operatorTotalUtilizedSD - operatorRecordedUtilizedSD // difference between the two is interest

interestPaid = Math.min(SDRequestedLeft, interestAccrued);
sdRequestedLeft -= interestPaid;
//Update repay amount in SDUtilityPool by transferring interestPaid tokens. Might have to update both utilized and bonded positions.

if (sdRequestedLeft == 0) { // You are done.}

utilizedPositionRepayAmount = Math.min(sdRequestedLeft, operatorRecordedUtilizedSD)
sdRequestedLeft -= utilizedPositionRepayAmount;
//definitely decrease the RecordedUtilizedSD & increment the Self Bonded SD & transfer utilizedPositionRepayAmount tokens to SD utilization pool.


if (sdRequestedLeft == 0) { // You are done}

// At this stage all interest + utilized amount has been repaid. The rest can to operator wallet & only self bonded position will be decremented
```

Will optimize this further.

Reply:
```
SDRequestedLeft = SDRequested;
operatorTotalUtilizedSD = utilityPool.currentUtilizedAmount(operator);

repayAmount = Math.min(SDRequestedLeft, operatorTotalUtilizedSD);
sdRequestedLeft -= repayAmount;
// transfer repayAmount tokens to SD utility pool. Decrement total utilized balance on pool & operator current utilized balance on pool.
interestPaid = Math.min(repayAmount, operatorTotalUtilizedSD - operatorRecordedUtilizedSD); // How do we make sure this math works for incentivized lending when the 2nd term might be negative?
utilizedAmountPaid = repayAmount - interestPaid;
operatorRecordedUtilizedSD -= utilizedAmountPaid;
SDBalances[operator] += utilizedAmountPaid;

if (sdRequestedLeft > 0) {
// Send sdRequestedLeft tokens to operator wallet.
}

return (interestPaid, utilizedAmountPaid, sdRequestedLeft);
```

Reply:
Is the above simpler?

Reply:
I have pushed another implementation of this, mostly the same as above, with a change that the repay function will return the actual repayment amount.